Modules/Common/AptosLikeMainModule.php

```diff
@@ -0,0 +1,297 @@
+                                {
+                                    $diff = bcadd($diff, $fee);
+                                }
+                                else
+                                {
+                                    $diff = bcsub($diff, $fee);
+                                }
+                            }
+
+                            // Check APT diff without fee
+                            if ($diff === '0')
+                            {
+                                continue;
+                            }
+
+                            $diff_sum = bcadd($diff_sum, $diff);
+                            $events[] = [
+                                'block' => $block['block_height'],
+                                'transaction' => $trx['hash'],
+                                'time' => $this->block_time,
+                                'address' => $address,
+                                'sort_key' => $sort_key++,
+                                'effect' => $diff,
+                                'failed' => $failed,
+                                'extra' => null,
+                            ];
+                        }
+
+                        if ($diff_sum !== '0')
+                        {
+                            $events[] = [
+                                'block' => $block['block_height'],
+                                'transaction' => $trx['hash'],
+                                'time' => $this->block_time,
+                                'address' => 'the-contract',
+                                'sort_key' => $sort_key++,
+                                'effect' => bcmul($diff_sum, '-1'),
+                                'failed' => $failed,
+                                'extra' => null,
+                            ];
+                        }
+                    }
+
+                    // Process transaction Fee
+                    $events[] = [
+                        'block' => $block['block_height'],
+                        'transaction' => $trx['hash'],
+                        'time' => $this->block_time,
+                        'address' => $trx['sender'],
+                        'sort_key' => $sort_key++,
+                        'effect' => '-' . $fee,
+                        'failed' => $failed,
+                        'extra' => 'f',
+                    ];
+
+                    $events[] = [
+                        'block' => $block['block_height'],
+                        'transaction' => $trx['hash'],
+                        'time' => $this->block_time,
+                        'address' => 'the-void',
+                        'sort_key' => $sort_key++,
+                        'effect' => $fee,
+                        'failed' => $failed,
+                        'extra' => 'f',
+                    ];
+
+                    break;
+
+                default:
+                    break;
```

Comment:
The default should be throwing a `ModuleError`. If some types are not processed, they should be explicitly listed.

Reply:
Fixed.